website/blog/2025-07-07-segfault-dependency-update.mdx

```diff
@@ -0,0 +1,81 @@
+---
+slug: segfault-dependency-update
+title: "SEGFAULT due to Dependency Update"
+authors: [majetideepak, czentgr]
+tags: [tech-blog, build]
+---
+
+## Background
+Velox depends on several [libraries](https://github.com/facebookincubator/velox/blob/main/CMake/resolve_dependency_modules/README.md).
+Some of these dependencies include open-source libraries from Meta, including [Folly](https://github.com/facebook/folly) and
+[Facebook Thrift](https://github.com/facebook/fbthrift). These libraries are in active development. These libraries also depend on each other, and hence they all have to be updated to the same version at the same time.
+
+Updating these dependencies typically involves modifying the Velox code to align with any public API or semantic changes in these dependencies.
+However, a recent upgrade of Folly and Facebook Thrift to version *v2025.04.28.00* caused a *SEGFAULT* only in one unit test in Velox
+named *velox_functions_remote_client_test*.
+
+## Investigation
+We immediately put on our *gdb* gloves and looked at the stack traces. This issue was also reproducible in a debug build.
```

Comment:
Looking at this on the preview I think readability would benefit by splitting this paragraph in two.